moonstone/filtering/basics_filtering.py

```diff
@@ -61,3 +63,50 @@ def filter(self) -> pd.DataFrame:
             return self._select_names()
         else:
             return self._exclude_names()
+
+
+class NaNPercentageFiltering(BothAxisFiltering):
+
+    def __init__(self, dataframe: pd.DataFrame, percentage_nan_allowed: Union[int, float] = 80, axis: int = 0):
+        """
+        :param percentage_nan_allowed: maximum percentage of NaN values allowed (between 0 and 100)
+        :param axis: axis to apply filtering (index (0) or columns(1))
+        """
+        self.percentage_nan_allowed = percentage_nan_allowed / 100
+        super().__init__(dataframe, axis=axis)
+
+    def filter(self) -> pd.DataFrame:
+        if self.axis == 0:
+            return self.df[self.df.isnull().mean(axis=1) <= self.percentage_nan_allowed]
+        return self.df.loc[:, self.df.isnull().mean() <= self.percentage_nan_allowed]
+
+
+class NumberOfDifferentValuesFiltering(BothAxisFiltering):
+    def __init__(self, dataframe: pd.DataFrame, min_number_values: int = 2,
```

Comment:
Maybe we could have also the option to do the opposite like only keep those with maximum n values?

Reply:
Good idea!